lib/class-core.php

```diff
@@ -322,8 +366,53 @@ public function admin_redirect() {
 
 		$site = $this->get_default_site();
 		if ( ! empty( $site->name ) && false === strpos( $_SERVER['HTTP_HOST'], $site->name ) ) { // WPCS: sanitization ok.
-			wp_redirect( esc_url_raw( 'http://' . $site->name . $_SERVER['REQUEST_URI'] ) ); // WPCS: sanitization ok.
+			wp_safe_redirect( esc_url_raw( sprintf( 'http%s://%s%s', is_ssl() ? 's' : '', $site->name, $_SERVER['REQUEST_URI'] ) ) ); // WPCS: sanitization ok.
```

Comment:
The `sprintf` call here seems to be used twice, maybe it can be pulled into a separate function, like `get_redirect_url`?

```sprintf( 'http%s://%s%s', is_ssl() ? 's' : '', $site->name, $_SERVER['REQUEST_URI'] ) )```